src/storage/src/source/mod.rs

```diff
@@ -408,9 +411,55 @@ pub(crate) trait SourceReader {
     ///
     /// Note that implementers are required to present messages in strictly ascending\
     /// offset order within each partition.
+    async fn next(
+        &mut self,
+    ) -> Option<Result<SourceMessage<Self::Key, Self::Value>, SourceReaderError>> {
+        // Compatiblity implementation that delegates to the deprecated [Self::get_next_method]
+        // call. Once all source implementations have been transitioned to implement
+        // [SourceReader::next] directly this provided implementation should be removed and the
+        // method should become a required method.
+        loop {
+            match self.get_next_message() {
+                Ok(NextMessage::Ready(msg)) => return Some(Ok(msg)),
+                Err(err) => return Some(Err(err)),
+                // There was a temporary hiccup in getting messages, check again asap.
+                Ok(NextMessage::TransientDelay) => {
+                    tokio::time::sleep(Duration::from_millis(1)).await
+                }
+                // There were no new messages, check again after a delay
+                Ok(NextMessage::Pending) => tokio::time::sleep(Duration::from_secs(1)).await,
+                Ok(NextMessage::Finished) => return None,
+            }
+        }
+    }
+
+    /// Returns the next message available from the source.
+    ///
+    /// Note that implementers are required to present messages in strictly ascending\
+    /// offset order within each partition.
+    ///
+    /// # Deprecated
+    ///
+    /// Source implementation should implement the async [SourceReader::next] method instead.
     fn get_next_message(
         &mut self,
     ) -> Result<NextMessage<Self::Key, Self::Value>, SourceReaderError>;
+
```

Comment:
default impl should always return `Ok(Pending)` for now, so that people can implement just `next` and ignore this method